concent_api/core/admin.py

```diff
@@ -1,44 +1,56 @@
 from django.contrib import admin
+from django.db.models import Q
 
-from common.admin    import ModelAdminReadOnlyMixin
-from .models        import PendingResponse
-from .models        import StoredMessage
-from .models        import Subtask
+from common.admin import ModelAdminReadOnlyMixin
+from common.helpers import get_current_utc_timestamp
+from common.helpers import parse_timestamp_to_utc_datetime
+from .models import PendingResponse
+from .models import StoredMessage
+from .models import Subtask
 
+ACTIVE_STATE_NAMES = [x.name for x in Subtask.ACTIVE_STATES]
+PASSIVE_STATE_NAMES = [x.name for x in Subtask.PASSIVE_STATES]
 
-class ActivePassiveStateFilter(admin.SimpleListFilter):
 
-    title          = 'Active/passive state'
+class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
+    title = 'Active/passive/downloads state'
     parameter_name = 'active_passive_state'
 
     def lookups(self, request, model_admin):
         return (
-            ('active',  'Active'),
+            ('active', 'Active'),
             ('passive', 'Passive'),
+            ('active_or_downloads', 'Active or with active downloads')
         )
 
     def queryset(self, request, queryset):
         if self.value() == 'active':
-            active_state_names = [x.name for x in Subtask.ACTIVE_STATES]
-            return queryset.filter(state__in=active_state_names)
+            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
         elif self.value() == 'passive':
-            passive_state_names = [x.name for x in Subtask.PASSIVE_STATES]
-            return queryset.filter(state__in=passive_state_names)
+            return queryset.filter(state__in=PASSIVE_STATE_NAMES)
+        elif self.value() == 'active_or_downloads':
+            current_timestamp = get_current_utc_timestamp()
```

Comment:
What if the value is different than those three?

You should have an `else` and put `assert False` in it so that the code fails if someone adds a new filter and forgets to include it here.

Reply:
Using ```else``` pycharm warned me that code is unreachable so I put just ```assert False```

Reply:
You could also do:
``` python
assert self.value() in {'active', 'passive', 'active_or_downloads'}
```
at the beginning of the function. And that would actually be more readable, though more repetitive too.